amr-wind/wind_energy/ABL.cpp

```diff
@@ -65,12 +65,19 @@ void ABL::initialize_fields(int level, const amrex::Geometry& geom)
         temp.setVal(0.0);
     }
 
+    bool interp_fine_levels = false;
     for (amrex::MFIter mfi(density); mfi.isValid(); ++mfi) {
         const auto& vbx = mfi.validbox();
 
-        (*m_field_init)(
-            vbx, geom, velocity.array(mfi), density.array(mfi),
-            temp.array(mfi));
+        interp_fine_levels = (*m_field_init)(
+            vbx, geom, velocity.array(mfi), density.array(mfi), temp.array(mfi),
+            level);
+    }
+
+    if (interp_fine_levels) {
+        // Fill the finer levels with coarse data
```

Comment:
this will break if level==2 and Smagorinsky and viscosity=0 since you'll have no strainrate and therefore no eddy viscosity